mycroft/client/speech/hotword_factory.py

```diff
@@ -97,6 +107,104 @@ def found_wake_word(self, frame_data):
         return hyp and self.key_phrase in hyp.hypstr.lower()
 
 
+class PreciseHotword(HotWordEngine):
+    def __init__(self, key_phrase="hey mycroft", config=None, lang="en-us"):
+        super(PreciseHotword, self).__init__(key_phrase, config, lang)
+        self.update_freq = 24  # in hours
+        self.url_base = ('https://raw.githubusercontent.com/'
+                         'MycroftAI/precise-data/')
+        self.exe_name = 'precise-stream'
+
+        ww = Configuration.get()['listener']['wake_word']
+        model_name = ww.replace(' ', '-') + '.pb'
+        model_folder = expanduser('~/.mycroft/precise')
+        if not isdir(model_folder):
+            mkdir(model_folder)
+        model_path = join(model_folder, model_name)
+
+        exe_file = self.find_download_exe()
+        LOG.info('Found precise executable: ' + exe_file)
+        self.update_model(model_name, model_path)
+
+        args = [exe_file, model_path, '1024']
+        self.proc = Popen(args, stdin=PIPE, stdout=PIPE)
+        self.has_found = False
+        self.cooldown = 20
+        t = Thread(target=self.check_stdout)
+        t.daemon = True
+        t.start()
+
+    def find_download_exe(self):
+        exe_file = resolve_resource_file(self.exe_name)
+        if exe_file:
+            return exe_file
+        try:
+            if call(self.exe_name + ' < /dev/null', shell=True) == 0:
+                return self.exe_name
+        except OSError:
+            pass
+
+        import platform
+        import stat
+
+        def snd_msg(cmd):
+            """Send message to faceplate"""
+            Popen('echo "' + cmd + '" > /dev/ttyAMA0', shell=True)
+
+        arch = platform.machine()
+        exe_file = expanduser('~/.mycroft/precise/' + self.exe_name)
```

Comment:
This path is not one of the paths checked in resolve_resource_file() so the exe is downloaded each startup.

Reply:
Should be resolved t=now. Good catch!